src/ILLink.Shared/TrimAnalysis/RuntimeTypeHandleValue.cs

```diff
@@ -1,16 +1,18 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using ILLink.Shared.DataFlow;
 using ILLink.Shared.TypeSystemProxy;
-using MultiValue = ILLink.Shared.DataFlow.ValueSet<ILLink.Shared.DataFlow.SingleValue>;
 
 namespace ILLink.Shared.TrimAnalysis
 {
 	/// <summary>
 	/// This is the System.RuntimeTypeHandle equivalent to a <see cref="SystemTypeValue"/> node.
 	/// </summary>
-	record RuntimeTypeHandleValue : SingleValue
+	sealed record RuntimeTypeHandleValue : SingleValue
 	{
 		public RuntimeTypeHandleValue (in TypeProxy representedType) => RepresentedType = representedType;
```

Comment:
Maybe add an assert that the type is not `Nullable<T>`

Reply:
When I added an assert I ran into cases where ``System.Nullable`1`` wouldn't have a type parameter, so it wouldn't make a `NullableValue` and would instantiate a SystemTypeValue or RuntimeTypeHandleValue. To me a SystemTypeValue makes sense since there is nothing extra to track in those situations, but would it make more sense to have the `UnderlyingType` of `NullableValue`'s be optional?